### PR TITLE
Update to Got 9, rename `endpoint` option and require Node.js 8

### LINKED ISSUE #33: Use `got.create()`
Using this new API could potentially simplify the code considerably. See: https://github.com/sindresorhus/gh-got/pull/29 and the docs https://github.com/sindresorhus/got/blob/master/advanced-creation.md

### PR DESCRIPTION
**Pull request by itaisteinherz**
This PR is nearly identical to sindresorhus/gh-got#29, and resolves #33.

### CODE REVIEW

**Comment by rarkins** [comment]
Thanks @itaisteinherz, I think best to wait until the same code has been merged into `gh-got` and after that we'll merge this.

**Comment by sindresorhus** [comment]
It was just merged. See: https://github.com/sindresorhus/gh-got/commit/402f07d77ca83c5413579d8bf475bcbb2d4fefc3

File: index.js [hunk 1]
```diff
@@ -2,61 +2,42 @@
...
-		'user-agent': 'https://github.com/singapore/gl-got'
-	}, opts.headers);
-
-	if (opts.token) {
-		opts.headers['PRIVATE-TOKEN'] = opts.token;
-	}
-
-	const url = /^https?/.test(path) ? path : opts.endpoint + path;
+		token: process.env.GITLAB_TOKEN,
+		baseUrl: process.env.GITLAB_ENDPOINT ? process.env.GITLAB_ENDPOINT.replace(/[^/]$/, '$&/') : 'https://gitlab.com/api/v4/',
```
Comment (sindresorhus) [inline index.js]: This can be simplified to:

```js
baseUrl: process.env.GITLAB_ENDPOINT || 'https://gitlab.com/api/v4',
```

File: package.json [hunk 2]
```diff
@@ -32,15 +32,15 @@
     "utility"
   ],
   "dependencies": {
-    "got": "^7.0.0",
+    "got": "^9.0.0",
```
Comment (sindresorhus) [inline package.json]: Use `"got": "^9.1.0",`

File: package.json [hunk 3]
```diff
@@ -32,15 +32,15 @@
...
     "is-plain-obj": "^1.1.0"
   },
   "devDependencies": {
-    "ava": "0.20.0",
+    "ava": "^1.0.0-beta.6",
     "get-stream": "3.0.0",
```
Comment (sindresorhus) [inline package.json]: Upgrade to latest AVA and `get-stream`.

File: readme.md [hunk 4]
```diff
@@ -67,7 +67,7 @@ GitLab [access token](https://docs.gitlab.com/ee/api/README.html#personal-access
 
 Can be set globally with the `GITLAB_TOKEN` environment variable.
 
-### endpoint
+### baseUrl
 
 Type: `string`<br>
 Default: `https://gitlab.com/api/v3/`
```
Comment (sindresorhus) [inline readme.md]: `https://gitlab.com/api/v3/` => `https://gitlab.com/api/v3`

File: index.js [hunk 5]
```diff
@@ -2,61 +2,42 @@
...
-			err.message = `${err.response.body.message} (${err.statusCode})`;
+		if (options.stream) {
+			return next(options);
 		}
 
-		throw err;
-	});
-}
+		return next(options).catch(err => {
+			if (err.response && isPlainObj(err.response.body)) {
```
Comment (sindresorhus) [inline index.js]: This can be simplified to `if (err.response && err.response.body) {`

**Comment by itaisteinherz** [comment]
Done 👌🏻 Thanks for the review @sindresorhus 🙂

**Pull request merged**